File: day01/day01.jl
using CSV

function go(filename::String)
    data = CSV.Tables.matrix(CSV.File(filename, delim=" ", header=false, ignorerepeated=true))
    println("Filename = $(filename)")
    data[:,1] = sort(data[:,1]) # sort! doesn't work for slices?   
    data[:,2] = sort(data[:,2])

    # part 1
    println("Total distance = $(sum(abs.(data[:,1] - data[:,2])))")

    # part 2
    score = 0
    for n in data[:,1]
        score += sum(data[:,2] .== n)*n
    end
    println("Similarity score = $(score)")
end

go("day01/test.txt")
go("day01/input.txt")

File: day01/day01.py
import numpy as np


def go(filename: str):
    loc_id = np.loadtxt(filename, dtype=np.int32)
    loc_id[:, 0] = np.sort(loc_id[:, 0])
    loc_id[:, 1] = np.sort(loc_id[:, 1])
    # part 1
    print(
        f"file = {filename}, total distance = {np.sum(np.abs(loc_id[:,0] - loc_id[:,1]))}"
    )

    # part 2
    sim_score = 0
    for n in loc_id[:, 0]:
        sim_score += np.sum(loc_id[:, 1] == n) * n
    print(f"file = {filename}, simularity score = {sim_score}")


go("day01/test.txt")
go("day01/input.txt")
